### PR TITLE
Attempt to fix compiling on arm64

### PR DESCRIPTION
**Pull request by robd003**
Attempt to resolve https://github.com/RedisGears/RedisGears/issues/839

### CODE REVIEW

**Comment by gkorland** [comment]
please format your code with `cargo fmt`

File: src/redismodule.rs [hunk 1]
```diff
@@ -109,9 +109,9 @@ impl RedisString {
     }
 
     #[allow(clippy::not_unsafe_ptr_arg_deref)]
-    pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[u8]) -> Self {
+    pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[c_char]) -> Self {
         let inner = unsafe {
-            raw::RedisModule_CreateString.unwrap()(ctx, s.as_ptr() as *const i8, s.len())
+            raw::RedisModule_CreateString.unwrap()(ctx, s.as_ptr() as *const c_char, s.len())
```
Comment (gkorland) [inline src/redismodule.rs:114]:
```suggestion
            raw::RedisModule_CreateString.unwrap()(ctx, s.as_ptr().cast::<c_char>(), s.len())
```

File: src/redismodule.rs [hunk 2]
```diff
@@ -109,9 +109,9 @@ impl RedisString {
     }
 
     #[allow(clippy::not_unsafe_ptr_arg_deref)]
-    pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[u8]) -> Self {
+    pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[c_char]) -> Self {
```
Comment (gkorland) [inline src/redismodule.rs:112]:
```suggestion
    pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[u8]) -> Self {
```

**Review by gkorland — CHANGES_REQUESTED** [review]

**Comment by robd003** [comment]
Currently stuck with more type mismatches
```
   Compiling redis-module v99.99.99 (/Users/robd/code/redismodule-rs)
error[E0308]: mismatched types
   --> src/raw.rs:564:54
    |
564 |         .map(|s| RedisString::create_from_slice(ctx, s))
    |                  ------------------------------      ^ expected slice `[u8]`, found `&[i8]`
    |                  |
    |                  arguments to this function are incorrect
    |
    = note: expected reference `&[u8]`
               found reference `&&[i8]`
note: associated function defined here
   --> src/redismodule.rs:112:12
    |
112 |     pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[u8]) -> Self {
    |            ^^^^^^^^^^^^^^^^^                                --------

error[E0308]: mismatched types
   --> src/context/mod.rs:211:63
    |
211 |             .map(|s| RedisString::create_from_slice(self.ctx, s))
    |                      ------------------------------           ^ expected slice `[u8]`, found `&[i8]`
    |                      |
    |                      arguments to this function are incorrect
    |
    = note: expected reference `&[u8]`
               found reference `&&[i8]`
note: associated function defined here
   --> src/redismodule.rs:112:12
    |
112 |     pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[u8]) -> Self {
    |            ^^^^^^^^^^^^^^^^^                                --------

error[E0308]: mismatched types
   --> src/context/mod.rs:236:80
    |
236 |         self.call_internal(command, options.options.as_ptr() as *const c_char, args)
    |              ------------- arguments to this function are incorrect            ^^^^ expected `i8`, found `u8`
    |
    = note: expected reference `&[&[i8]]`
               found reference `&[&[u8]]`
note: associated function defined here
   --> src/context/mod.rs:203:12
    |
203 |     pub fn call_internal(
    |            ^^^^^^^^^^^^^
...
207 |         args: &[&[c_char]],
    |         ------------------

error[E0308]: mismatched types
   --> src/context/mod.rs:243:13
    |
240 |         self.call_internal(
    |              ------------- arguments to this function are incorrect
...
243 |             &args.iter().map(|v| v.as_bytes()).collect::<Vec<&[u8]>>(),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected slice `[&[i8]]`, found struct `Vec`
    |
    = note: expected reference `&[&[i8]]`
               found reference `&Vec<&[u8]>`
note: associated function defined here
   --> src/context/mod.rs:203:12
    |
203 |     pub fn call_internal(
    |            ^^^^^^^^^^^^^
...
207 |         args: &[&[c_char]],
    |         ------------------

error[E0308]: mismatched types
   --> src/context/mod.rs:595:50
    |
595 |         RedisString::create_from_slice(self.ctx, s)
    |         ------------------------------           ^ expected `u8`, found `i8`
    |         |
    |         arguments to this function are incorrect
    |
    = note: expected reference `&[u8]`
               found reference `&[i8]`
note: associated function defined here
   --> src/redismodule.rs:112:12
    |
112 |     pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[u8]) -> Self {
    |            ^^^^^^^^^^^^^^^^^                                --------

For more information about this error, try `rustc --explain E0308`.
error: could not compile `redis-module` due to 5 previous errors
```

**Comment by gkorland** [comment]
Fixed in https://github.com/RedisLabsModules/redismodule-rs/commit/cd09e7d02ebba9ccdc989f388612fecd21441ff6

**Pull request closed without merging**